internal/action/team_access.go

```diff
@@ -0,0 +1,58 @@
+package action
+
+import (
+	"fmt"
+
+	"github.com/takescoop/terraform-cloud-workspace-action/internal/tfeprovider"
+)
+
+// TeamAccessInput is a list of team access settings
+type TeamAccessInput []TeamAccessInputItem
+
+// TeamAccessInputItem represents a single team access setting
+type TeamAccessInputItem struct {
+	Access      string                      `yaml:"access,omitempty"`
+	Permissions *TeamAccessPermissionsInput `yaml:"permissions,omitempty"`
+	TeamName    string                      `yaml:"name"`
+	TeamID      string                      `yaml:"id"`
+
+	// TODO: remove this, objects should not store data on behalf of callers
+	WorkspaceName string
```

Comment:
The alternative would be to add the appropriate value to tfeprovider.TeamAccess directly at a later execution time then add the ws name here for later use? Or take it as an argument to ToResource?

Reply:
Team accesses is a child resource of a workspace, or multiple workspaces. I found this confusing when I first spotted it, only to realize the field is holding data for later use in imports. This object should have one purpose—to hold a team access input entry. If it needs to be combined with other data later, that should be a new type. Other functions should not depend on `TeamAccessInput` to define fields where they can put related data.

```go
type TeamAccessImport struct {
  Input TeamAccessInput
  Workspace string
}
```

Reply:
i see, a new type makes sense. fwiw it's used for imports and also for defining the ref for the `workspace_id`